kolibri/plugins/coach/assets/src/composables/useFetchContentNode.js

```diff
@@ -0,0 +1,40 @@
+import { ref } from 'vue';
+import ContentNodeResource from 'kolibri-common/apiResources/ContentNodeResource';
+
+export default function useFetchContentNode(contentId) {
+  const contentNode = ref({});
+  const ancestors = ref([]);
+  const questions = ref([]);
+  const loading = ref(true);
+
+  const fetchContentNode = async () => {
+    ContentNodeResource.fetchModel({
+      id: contentId,
+      getParams: { no_available_filtering: true },
+    }).then(node => {
+      if (node != null) {
+        loading.value = false;
+        contentNode.value = node;
+
+        if (node.ancestors.length) {
+          ancestors.value = node.ancestors;
+        }
+
+        if (node.assessmentmetadata) {
+          questions.value = node.assessmentmetadata.assessment_item_ids;
+        }
+      } else {
+        loading.value = false;
+      }
+    });
+  };
+
+  fetchContentNode();
```

Comment:
Non-blocking note. You could skip making the function `fetchContentNode` altogether and just run the code within it directly since you're not returning the function from the composable and you call it within the comopsable.